src/trainer.py

```diff
@@ -101,8 +99,15 @@ def evaluate(self, loss_f, metric_fs={}, mode='dev'):
             metric = fn(binary_pred, mask)
             epoch_metrics[name] += metric
 
-    return (epoch_loss / n), {name: value/n for name, value in epoch_metrics.items()}
-  
+        if self.config.store_images and i % 10  == 0:
+          img, pred, mask = img.median(axis=1).cpu(), pred.cpu(), mask.cpu()
+          for j in range(img.shape[0]):
+            merged = np.concatenate([img[j], pred[j], mask[j]], axis=1)
+            wandb_imgs.append(wandb.Image(merged))
+
+    wandb.log({f"{mode}_images": wandb_imgs})
```

Comment:
This logs the image (with all 10/12 channels), the original mask and the logits
I think the best would be rgb (or [B5, B4, B2]), and the actual segmentation map (after Sigmoid or Softmax)

Reply:
Hm, I thought I had taken care of this by taking the mean over channels in the previous two lines, but you're right, things aren't quite right. I'm looking into it.

Reply:
Yep, I just needed to apply the activations. I'm also taking three bands now, though you can't specify them in advance.

We should check to see if the activations are saturating too early in training. I'm noticing large variance, but am not sure if it's normal yet.